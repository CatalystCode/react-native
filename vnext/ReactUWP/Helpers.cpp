#include "pch.h"
#include "Helpers.h"
#include <Modules/NativeUIManager.h>

namespace react { namespace uwp {

ReactId getReactId(IReactInstance *instance, winrt::FrameworkElement const& fe)
{
  ReactId reactId;
  assert(instance);
  if (auto uiManager = static_cast<NativeUIManager*>(instance->NativeUIManager()))
  {
    if (auto peer = uiManager->reactPeerOrContainerFrom(fe))
    {
      reactId.isValid = true;
      reactId.tag = GetTag(peer);
    }
  }
  return reactId;
};

}};
