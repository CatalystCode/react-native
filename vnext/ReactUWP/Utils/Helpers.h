// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#pragma once

#include <stdint.h>
#include <winrt/Windows.UI.Xaml.h>
#include <IReactInstance.h>
#include <folly/dynamic.h>

namespace winrt {
  using namespace Windows::UI::Xaml;
}

namespace react { namespace uwp {

  using namespace std;

  struct ReactId {
    int64_t tag{ 0 };
    bool isValid{ false };
  };

  template<typename T>
  inline typename T as(folly::dynamic const& obj)
  {
    return (T)(obj.asInt());
  }

  template<>
  inline typename std::string as<std::string>(folly::dynamic const& obj)
  {
    return obj.asString();
  }

  template<>
  inline typename bool as<bool>(folly::dynamic const& obj)
  {
    return obj.asBool();
  }

  template<>
  inline typename double as<double>(folly::dynamic const& obj)
  {
    return obj.asDouble();
  }

   ReactId getReactId(IReactInstance *instance, winrt::FrameworkElement const& fe);
}}
