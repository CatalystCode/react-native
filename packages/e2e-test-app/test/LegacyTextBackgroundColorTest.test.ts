/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import {goToComponentExample, dumpVisualTree} from './Helpers';

beforeAll(async () => {
  await goToComponentExample('LegacyTextBackgroundColorTest');
});

describe('LegacyTextBackgroundColorTest', () => {
  /* Test case #1: view and image displayed with no border and cornerRadius */
  test('TextBackgroundColorTest', async () => {
    const dump = await dumpVisualTree('text-color-background-view');
    expect(dump).toMatchSnapshot();
  });
});
