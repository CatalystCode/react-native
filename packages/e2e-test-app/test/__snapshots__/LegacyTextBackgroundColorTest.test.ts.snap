// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegacyTextBackgroundColorTest TextBackgroundColorTest 1`] = `
Object {
  "AutomationId": "text-color-background-view",
  "Background": null,
  "BorderBrush": "#00000000",
  "BorderThickness": "0,0,0,0",
  "Clip": null,
  "CornerRadius": "0,0,0,0",
  "FlowDirection": "LeftToRight",
  "Height": 247,
  "HorizontalAlignment": "Stretch",
  "Left": 0,
  "Margin": "0,0,0,0",
  "Top": 0,
  "VerticalAlignment": "Stretch",
  "Visibility": "Visible",
  "Width": 800,
  "XamlType": "Microsoft.ReactNative.ViewPanel",
  "children": Array [
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FFFFC0CB",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer pink",
      "TextHighlighters": Array [],
      "Top": 0,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer red nested",
      "TextHighlighters": Array [],
      "Top": 19,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer nested white on blue",
      "TextHighlighters": Array [
        Object {
          "Background": "#FF0000FF",
          "Ranges": Array [
            Object {
              "Length": 20,
              "StartIndex": 6,
            },
          ],
        },
      ],
      "Top": 38,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FFFFC0CB",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer pink nested red",
      "TextHighlighters": Array [],
      "Top": 57,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer on green nested white on inherit green",
      "TextHighlighters": Array [
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 14,
              "StartIndex": 0,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 14,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 29,
              "StartIndex": 15,
            },
          ],
        },
      ],
      "Top": 76,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FFFFA500",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer orange on green nested white on blue",
      "TextHighlighters": Array [
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 21,
              "StartIndex": 0,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 21,
            },
          ],
        },
        Object {
          "Background": "#FF0000FF",
          "Ranges": Array [
            Object {
              "Length": 20,
              "StartIndex": 22,
            },
          ],
        },
      ],
      "Top": 95,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FFFFA500",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer orange nested white on blue",
      "TextHighlighters": Array [
        Object {
          "Background": "#FF0000FF",
          "Ranges": Array [
            Object {
              "Length": 20,
              "StartIndex": 13,
            },
          ],
        },
      ],
      "Top": 114,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FFFFA500",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "nested orange inherit on blue",
      "TextHighlighters": Array [
        Object {
          "Background": "#FF0000FF",
          "Ranges": Array [
            Object {
              "Length": 29,
              "StartIndex": 0,
            },
          ],
        },
      ],
      "Top": 133,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer no_color START_NESTED green DEEPER_NESTED blue END_NESTED attributes.",
      "TextHighlighters": Array [
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 18,
              "StartIndex": 15,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 33,
            },
          ],
        },
        Object {
          "Background": "#FF0000FF",
          "Ranges": Array [
            Object {
              "Length": 18,
              "StartIndex": 34,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 52,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 10,
              "StartIndex": 53,
            },
          ],
        },
      ],
      "Top": 152,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer no_color START_NESTED no_color DEEPER_NESTED blue END_NESTED attributes.",
      "TextHighlighters": Array [
        Object {
          "Background": "#FF0000FF",
          "Ranges": Array [
            Object {
              "Length": 18,
              "StartIndex": 37,
            },
          ],
        },
      ],
      "Top": 171,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer no_color START_NESTED green DEEPER_NESTED inherit green END_NESTED attributes.",
      "TextHighlighters": Array [
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 19,
              "StartIndex": 15,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 27,
              "StartIndex": 34,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 61,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 10,
              "StartIndex": 62,
            },
          ],
        },
      ],
      "Top": 190,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer red START_NESTED inherit red DEEPER_NESTED inherit red END_NESTED attributes.",
      "TextHighlighters": Array [
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 9,
              "StartIndex": 0,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 9,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 25,
              "StartIndex": 10,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 25,
              "StartIndex": 35,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 60,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 10,
              "StartIndex": 61,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 71,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 11,
              "StartIndex": 72,
            },
          ],
        },
      ],
      "Top": 209,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "Outer red START_NESTED green DEEPER_NESTED blue END_NESTED attributes.",
      "TextHighlighters": Array [
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 9,
              "StartIndex": 0,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 9,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 18,
              "StartIndex": 10,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 28,
            },
          ],
        },
        Object {
          "Background": "#FF0000FF",
          "Ranges": Array [
            Object {
              "Length": 18,
              "StartIndex": 29,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 47,
            },
          ],
        },
        Object {
          "Background": "#FF008000",
          "Ranges": Array [
            Object {
              "Length": 10,
              "StartIndex": 48,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 1,
              "StartIndex": 58,
            },
          ],
        },
        Object {
          "Background": "#FFFF0000",
          "Ranges": Array [
            Object {
              "Length": 11,
              "StartIndex": 59,
            },
          ],
        },
      ],
      "Top": 228,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 800,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
  ],
}
`;
